packages/@expo/cli/src/start/doctor/dependencies/validateDependenciesVersions.ts

```diff
@@ -32,7 +34,12 @@ export async function validateDependenciesVersionsAsync(
   exp: Pick<ExpoConfig, 'sdkVersion'>,
   pkg: PackageJSONConfig,
   packagesToCheck?: string[]
-): Promise<boolean> {
+): Promise<boolean | null> {
+  if (APISettings.isOffline) {
```

Comment:
I added the offline check here, but since `expo start` is the only one supporting the `--offline` flag, we could also not-run this function if running in offline mode.

No strong opinions on either one, but with this, we know that this won't pop up again if used in other commands.